I was one of those affected by issue #9660.

I tried this PR and it indeed fixes the crash.

There is one new small annoyance though. On Windows (at least), I now get link errors unless I add opengl32.lib to my project, which was not required when this was a dynamic library. I'm not sure if this is to be expected or if there is a way to automatically link to opengl32.lib too using vcpkg. I tested with VS2019 in Debug x64 configuration.

Sample of link errors:
```
1>ImGui-SFML.lib(imgui-SFML.cpp.obj) : error LNK2019: unresolved external symbol __imp_glBindTexture referenced in function "void __cdecl `anonymous namespace'::RenderDrawLists(struct ImDrawData *)" (?RenderDrawLists@?A0x823403ac@@YAXPEAUImDrawData@@@Z)
1>ImGui-SFML.lib(imgui-SFML.cpp.obj) : error LNK2019: unresolved external symbol __imp_glBlendFunc referenced in function "void __cdecl `anonymous namespace'::RenderDrawLists(struct ImDrawData *)" (?RenderDrawLists@?A0x823403ac@@YAXPEAUImDrawData@@@Z)
1>ImGui-SFML.lib(imgui-SFML.cpp.obj) : error LNK2019: unresolved external symbol __imp_glColorPointer referenced in function "void __cdecl `anonymous namespace'::RenderDrawLists(struct ImDrawData *)" (?RenderDrawLists@?A0x823403ac@@YAXPEAUImDrawData@@@Z)
```

--- a/ports/imgui-sfml/portfile.cmake
+++ b/ports/imgui-sfml/portfile.cmake
@@ -1,5 +1,3 @@
-string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" SFML_STATIC)
-
 vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 
 vcpkg_from_github(
@@ -14,6 +12,18 @@ vcpkg_from_github(
         0003-fix_osx.patch
         004-fix-find-sfml.patch
 )
+
+if (VCPKG_TARGET_IS_WINDOWS)
+    file(GLOB SFML_DYNAMIC_LIBS "${CURRENT_INSTALLED_DIR}/bin/sfml-*")
+else()
+    file(GLOB SFML_DYNAMIC_LIBS "${CURRENT_INSTALLED_DIR}/bin/libsfml-*")
+endif()
+
+if (SFML_DYNAMIC_LIBS)
+    set(SFML_STATIC OFF)
+else()
+    set(SFML_STATIC ON)
+endif()
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}

